### PR TITLE
Add exclude of directory "db_analyze" to generate template to gpexpand

### PR DESCRIPTION
**Pull request by red1452**
If run `analyze_db` for any database, than for this database folder "db_analyze" is created. It may contain large volume of data. This catalog is useless for segments. If after it create new segment using `gpexpand`, than folder "db_analyze" will be copied to new segment. This folder is useless for new segment. Because of this exclude "db_analyze" is added to `PgBaseBackup`.

Steps to reproduce the problem:
1) Create demo cluster.
2) Run `analyzedb -d postgres -a`.
3) Create config file "expand.conf" for `gpexpand`:
```
sdw1|sdw1|36101|<PATH_TO_GPDB_SRC>/gpAux/gpdemo/datadirs/dbfast4|9|3|p
sdw1|sdw1|36102|<PATH_TO_GPDB_SRC>/gpAux/gpdemo/datadirs/dbfast_mirror4|10|3|m
```
4) Run `gpexpand -i expand.conf`.
5) Done.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/bin/gppylib/commands/pg.py [hunk 1]
```diff
@@ -320,6 +320,8 @@ def __init__(self, target_datadir, source_host, source_port, create_slot=False,
             cmd_tokens.append('./db_dumps')
             cmd_tokens.append('-E')
             cmd_tokens.append('./promote')
+            cmd_tokens.append('-E')
```
Comment (ashwinstar) [inline gpMgmt/bin/gppylib/commands/pg.py:323]: Adding here is not just going to exclude it for gpexpand but also for full mirror creation. Do we wish to loose this directory in failover the mirror scenario after full recovery of the mirror?

Comment (red1452) [reply gpMgmt/bin/gppylib/commands/pg.py:323]: Catalog "db_analyze" is useless for segments and their mirrors. It needs only for master which execute `analyzedb` to collect some information about database. I think, data at catalog "db_analyze" is actual only for master, which execute `analyzedb`. If you want to migrate master to new host, you should execute `analyzedb` at new master after migration process.

Comment (red1452) [reply gpMgmt/bin/gppylib/commands/pg.py:323]: I think, that data, which is not update at standby, will not be useful after master is failing. After it this data should update. This data remains relevant in time after executing analyze before any change of data at the database.

**Review by ashwinstar — APPROVED** [review]
LGTM

**Review by shrakesh — APPROVED** [review]
did some manual testing to verify the scenario related to movemirrors and gpexpand. 
LGTM

**Comment by shrakesh** [comment]
@red1452 can you please rebase/resolve the conflicts and push it to merge?

**Comment by red1452** [comment]
> @red1452 can you please rebase/resolve the conflicts and push it to merge?

Done.

**Pull request merged**